test/pylib/cpp/common_cpp_conftest.py

```diff
@@ -71,13 +71,19 @@ def get_disabled_tests(config: dict, modes: list[str]) -> dict[str, set[str]]:
 
 def read_suite_config(directory: Path) -> dict[str, str]:
     """
-    Helper method that will return the configuration from the suite.yaml file
+    Helper method that returns the configuration from the test_config.yaml file.
+    It can be that there's no test_config.yaml that means there are no additional parameters for Scylla itself or no
+    filtering on what modes tests should be executed or skipped
     """
-    with open(directory / 'suite.yaml', 'r') as cfg_file:
-        cfg = yaml.safe_load(cfg_file.read())
-        if not isinstance(cfg, dict):
-            raise RuntimeError('Failed to load tests: suite.yaml is empty')
-        return cfg
+    config = directory / 'test_config.yaml'
+    if config.exists() and os.stat(config).st_size != 0:
+        with open(directory / 'test_config.yaml', 'r') as cfg_file:
+            cfg = yaml.safe_load(cfg_file.read())
+            if not isinstance(cfg, dict):
+                raise ValueError(f"Invalid test_config.yaml format in {directory}. Expected a dictionary, got {type(cfg)}")
```

Comment:
I wanna nag a little:

1. It's better to define a constant for `test_config.yaml` string somewhere.
2. `config` is an instance of `Path`, so, you can use methods of this class instead of using `os.stat()` and `open()`
3. You re-create same `Path` instance again on opening a file. Just can re-use `config`, I think.
4. `r` is a default mode on file open, can be avoided
5. Need to set encoding for text modes. Please, add `encoding='utf-8'` to parameters.
6. Comparing with 0 is not idiomatic Python code, you can use just `... and <>.st_size` here.
7. Finally, `Path` has `.read_text()` method. You can use it instead of `with ... as file : ... file.read()`. Just don't forget to add encoding parameter.